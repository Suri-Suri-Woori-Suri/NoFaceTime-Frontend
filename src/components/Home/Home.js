import React from 'react';
import styles from './Home.module.css';
import { BACKGROUND_IMG } from '../../constants';

const Home = () => {
  return (
    <div className={styles.Home}>
      <div className={styles.ServiceDescription}>
        <p>
          <b>Enjoy</b>
          <br />
          your vedeo meeting
          <br />
          with emojis
        </p>
      </div>
      <div className={styles.ServiceImage}>
        <img
          src={BACKGROUND_IMG.NOTEBOOK}
          alt="notebook"
          className={styles.NoteBook} />
      </div>
    </div>
  );
};

export default Home;
