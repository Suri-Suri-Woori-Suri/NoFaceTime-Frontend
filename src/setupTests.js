import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

export const renderWithRouter = component => render(<Router>{component}</Router>);
export const mountWithRouter = component => mount(<Router>{component}</Router>);

configure({ adapter: new Adapter() });
