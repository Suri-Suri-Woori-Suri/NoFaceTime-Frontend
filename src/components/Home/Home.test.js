import React from 'react';
import { cleanup } from '@testing-library/react';

import { mountWithRouter, renderWithRouter } from '../../setupTests';
import Home from './Home';

describe('<Home />', () => {
  afterEach(cleanup);

  const wrapper = mountWithRouter(<Home />);

  it('renders text properly in main page', () => {
    const { getByText } = renderWithRouter(<Home />);

    expect(getByText('Are you there?')).toBeInTheDocument();
  });

  it(`render img properly to have altervative text as 'notebook'`, () => {
    const { getByAltText } = renderWithRouter(<Home />);

    expect(getByAltText('notebook')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
