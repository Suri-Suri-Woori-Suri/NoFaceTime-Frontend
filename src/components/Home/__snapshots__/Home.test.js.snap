// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Home>
      <div
        className="Home"
      >
        <div
          className="ServiceDescription"
        >
          <p>
            <b>
              Are you there?
            </b>
            <br />
            Check your students
            <br />
            in real time through emoji
          </p>
        </div>
        <div
          className="ServiceImage"
        >
          <img
            alt="notebook"
            className="NoteBook"
            src="https://no-face-time.s3.ap-northeast-2.amazonaws.com/notebook.jpg"
          />
        </div>
      </div>
    </Home>
  </Router>
</BrowserRouter>
`;
